sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client_helpers.py

```diff
@@ -845,7 +848,7 @@ def _abort_copy_options(copy_id: Union[str, Dict[str, Any], BlobProperties], **k
     :rtype: Dict[str, Any]
     """
     access_conditions = get_access_conditions(kwargs.pop('lease', None))
-    if isinstance(copy_id, BlobProperties):
+    if isinstance(copy_id, BlobProperties) and copy_id.copy.id is not None:
```

Comment:
I think this may be an issue with the `BlobProperties` model? I would think `copy.id` would not be optional.

Reply:
Here `copy_id.copy` refers to a `CopyProperties` object. This model [states](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/azure/storage/blob/_models.py#L607):

> These properties will be `None` if this blob has never been the destination
    in a Copy Blob operation, or if this blob has been modified after a concluded
    Copy Blob operation, for example, using Set Blob Properties, Upload Blob, or Commit Block List.

And so I think these truly are `Optional`?

Reply:
`copy_id.copy` can be `None` but I think `copy.id` is not optional, meaning this None check does not really accomplish anything because if `copy_id.copy` is None, this will still throw an exception. I guess we assume that the `BlobProperties` being pass had a populated `copy` which I think is fine. Just remove this and ignore whatever.